dlclive/dlclive.py

```diff
@@ -478,3 +486,154 @@ def close(self):
+        display_radius: int = 3,
+        display_cmap: str = "bmy",
+    ):
+        if track_method not in ("box", "ellipse"):
+            raise ValueError("`track_method` should be either `box` or `ellipse`.")
+
+        self.model_path = model_path
+        super().__init__(
+            Path(model_path).parent,
+            resize=resize,
+            convert2rgb=convert2rgb,
+            processor=processor,
+            display=display,
+            pcutoff=pcutoff,
+            display_radius=display_radius,
+            display_cmap=display_cmap,
+        )
+        self.n_animals = n_animals
+        self.n_multibodyparts = n_multibodyparts
+        self.track_method = track_method
+        self.min_hits = min_hits
+        self.max_age = max_age
+        self.sim_threshold = sim_threshold
+
+    def read_config(self):
+        cfg_path = Path(self.path).resolve() / "pose_cfg.yaml"
+        if not cfg_path.exists():
+            raise FileNotFoundError(
+                f"The pose configuration file for the exported model at {str(cfg_path)} was not found. Please check the path to the exported model directory"
+            )
+
+        self.cfg = load_config(cfg_path)
+        self.cfg["batch_size"] = 1
+        self.cfg["init_weights"] = self.model_path.split(".")[0]
+        self.identity_only = self.cfg["num_idchannel"] > 0
+
+    def init_inference(
+        self,
+        frame=None,
+        allow_growth=False,
+        **kwargs,
+    ):
+        self.sess, self.inputs, self.outputs = predict.setup_pose_prediction(
+            self.cfg, allow_growth=allow_growth,
+        )
+
+        if self.track_method == "box":
+            self.mot_tracker = trackingutils.SORTBox(
+                self.max_age, self.min_hits, self.sim_threshold,
+            )
+        else:
+            self.mot_tracker = trackingutils.SORTEllipse(
+                self.max_age, self.min_hits, self.sim_threshold,
+            )
+
+
+        data = {
+            "metadata": {
+                "all_joints_names": self.cfg["all_joints_names"],
+                "PAFgraph": self.cfg["partaffinityfield_graph"],
+                "PAFinds": self.cfg.get("paf_best", np.arange(self.n_multibodyparts))
+            }
+        }
+        # Hack to avoid IndexError when determining _has_identity
+        temp = {"identity": []} if self.identity_only else {}
+        data["frame0"] = temp
+        self.ass = inferenceutils.Assembler(
+            data,
+            max_n_individuals=self.n_animals,
+            n_multibodyparts=self.n_multibodyparts,
+            greedy=True,  # TODO Benchmark vs optimal matching
+            identity_only=self.identity_only,
+            max_overlap=1,
+        )
+
+        if frame is not None:
+            pose = self.get_pose(frame, **kwargs)
+        else:
+            pose = None
+
+        self.is_initialized = True
+
+        return pose
+
+    def get_pose(self, frame=None, **kwargs):
+        if frame is None:
```

Comment:
iirc `frame` being optional is a hangover from an old version. I can't think of a reason to keep it that way?